micrometer-core/src/main/java/io/micrometer/core/instrument/Tags.java

```diff
@@ -149,13 +149,18 @@ public void remove() {
 
     }
 
+    @Override
+    public Spliterator<Tag> spliterator() {
+        return Spliterators.spliterator(tags, 0, last, Spliterator.IMMUTABLE | Spliterator.ORDERED
```

Comment:
I doubt it makes much of a difference, but I wonder if there is a case in which passing the range `[0, last]` results in something different than passing the underlying `tags` array without bounds. Looking at the underlying implementation, it would avoid a bounds check by only passing the array.

Reply:
Based on how the `dedup` method works, it looked to me like the number of elements may be less than the array length, is that not the case? If we passed the array without the start / end index, I think we'd potentially produce incorrect or duplicate results, right?

Reply:
Yes, you are right. I had forgotten (and didn't look closely enough at) what the `dedup` method does.